### ISSUE TITLE
`OPTIONS` only works with `CONANFILE`, not with `REQUIRES`

### ISSUE THREAD

**Issue by puetzk** [conan-io/cmake-conan#279]
It appears that the support for `OPTIONS` is nested inside of the support for `CONANFILE`
https://github.com/conan-io/cmake-conan/blob/2c98e07189eb62bf103cefa603f16d343ee4d75d/conan.cmake#L409
https://github.com/conan-io/cmake-conan/blob/2c98e07189eb62bf103cefa603f16d343ee4d75d/conan.cmake#L415-L418

It is not clear to me what's wrong with setting options for packages listed in `REQUIRES`, so presumably this should just be outside the if?

**Comment by czoido** [comment]
Hi @puetzk,
`OPTIONS` should also be set with no `CONANFILE` defined here: https://github.com/conan-io/cmake-conan/blob/2c98e07189eb62bf103cefa603f16d343ee4d75d/conan.cmake#L486-L489 in the temporary _conanfile.txt_ that is generated.

**Comment by puetzk** [comment]
Hmm. I missed that, but it actually didn't work, so now I wonder why. Perhaps I'm trying to override options that are in the profile from `$CONAN_DEFAULT_PROFILE_PATH`, maybe there's a difference in how options from the conanfile.txt rank vs options passed on the command-line?

**Comment by czoido** [comment]
Hi @puetzk,
Yes the command line options take preference over the rest, so that's why it's happening.
If your use case is for overriding options from the profiles you could actually try to use profile composition, you can have a look here: https://docs.conan.io/en/latest/reference/profiles.html#profile-composition 
If you specify multiple profiles in `conan_cmake_run` they should be composed:

```
conan_cmake_run(REQUIRES fmt/6.1.2
                BASIC_SETUP CMAKE_TARGETS
                BUILD missing
                PROFILE default
                PROFILE shared)
```

Would that help you?

**Comment by puetzk** [comment]
Hmm. I don't really want a global profile for these unusual sets of option overrids. But I assume PROFILE can be an absolute path (it can in most other places)? If so I don't love it, but should be able to make that work.

It does still seem a little weird that `OPTIONS` overrides the profile if you're using `CONANFILE`, but is overridden by the profile if you are using `REQUIRES`, though.

**Comment by czoido** [comment]
> It does still seem a little weird that `OPTIONS` overrides the profile if you're using `CONANFILE`, but is overridden by the profile if you are using `REQUIRES`, though.

Yes, you are right, It should always have the same behaviour, we will try to fix this for v0.16
